intermediate/code/6/hero.py

```diff
@@ -0,0 +1,52 @@
+class Hero:
+    M = 1.5
+    K = 2
+    N = 1.7
+
+    def __init__(self, name) -> None:
+        self.name = name
+        self.hp = 10
+        self.damage = 2
+        self.level = 1
+        self.coins = 0
+        
+    def heal(self) -> None: 
+        self.hp = self.hp * Hero.N
+        print(f'You healed yourself, now you have {self.hp} lives!')
+
+    def level_up(self) -> None:
+        if (self.coins / (self.level + 1)) >= Hero.K:
+            self.hp = 10 * Hero.M
+            self.damage = self.damage * Hero.M
+            self.level = self.level + 1
+
+            print(f'Leveled up! you are now in level {self.level}')
+
+    def attack(self, monster) -> None:
+        from monster import Monster
+        monster_lives = monster.reduce_health(self)
+        print('You attacked the monster!')
+
+        if monster_lives == 0:
+            self.coins = self.coins + self.level
+
+    def defend(self, damage) -> float:
+        return damage * 0.2
```

Comment:
functions of classes should belong to the class, this function is supposed to indicate that the hero has a defend for one turn, and then it gets canceled, not to use to update the perameter of the monster